packages/react/src/components/gdpr-consent-dialog/actions.tsx

```diff
@@ -48,6 +48,7 @@ export const Actions = ({
     try {
       const consents = await updateGdprConsent(consentLevels);
       onSuccess?.(consents);
+      dispatch({ type: "SET_CONSENT_SETTINGS", payload: consents });
```

Comment:
If there is a refactor on the cards I would like suggest not passing the `dispatch()` function reference into child components. This creates an unnecessary coupling, but my main issue is the added complexity this pattern brings.

It's quite difficult to reason about and took me some time to get my head around.

My suggestion: `<Actions>` should be presentational only, use "events" to signal user interaction and update state.

```ts
<Actions
  onAcceptAll={dispatch({ "ACCEPT_ALL", payload: defaultAcceptAllLevels })}
  onRejectAll={...}
  onCustomize={...}
  loading={...}
/>
```

But as I start to unwind the component like this, it doesn't feel like the abstraction adds very much value - it's mostly indirection.

Why don't we just move the four buttons into `<ConsentDialog>`?

Reply:
I agree with your point about passing `dispatch`, I will change it.
Regarding moving the buttons, I just feel like `<ConsentDialog>` will get bloated too much if it has those buttons, but if you feel like it will improve readability I will move them.